instrumentation/net/http/otelhttp/example/server/github.com/labstack/echo/server.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+
+	"github.com/labstack/echo/v4"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	stdout "go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	oteltrace "go.opentelemetry.io/otel/trace"
+)
+
+var tracer = otel.Tracer("echo-server")
+
+func main() {
+	tp, err := initTracer()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := tp.Shutdown(context.Background()); err != nil {
+			log.Printf("Error shutting down tracer provider: %v", err)
+		}
+	}()
+	r := echo.New()
+	r.Use(echo.WrapMiddleware(otelhttp.NewMiddleware("my-server")))
+
+	r.GET("/users/:id", func(c echo.Context) error {
+		id := c.Param("id")
+		name := getUser(c.Request().Context(), id)
+		return c.JSON(http.StatusOK, struct {
+			ID   string
+			Name string
+		}{
+			ID:   id,
+			Name: name,
+		})
+	}, echo.WrapMiddleware(otelhttp.WithRouteTagMiddleware("/users/:id")))
```

Comment:
Would it not be possible to get rid of the `"/users/:id"` duplication?

I do not like requiring to add `echo.WrapMiddleware(otelhttp.WithRouteTagMiddleware()` for each registering handler.

I would prefer to have something like


```go
	r.Use(echo.WrapMiddleware(otelhttp.NewMiddleware("my-server")))
        r.Use(OTelRouteTaggerMiddleware()) // or: r.Use(otelecho.RouteTaggerMiddleware())

	r.GET("/users/:id", func(c echo.Context) error {
		id := c.Param("id")
		name := getUser(c.Request().Context(), id)
		return c.JSON(http.StatusOK, struct {
			ID   string
			Name string
		}{
			ID:   id,
			Name: name,
		})
	})
```

Reply:
@pellared I can't come up with a way we can create a generic middleware for this since the route tag is usually router specific and not possible to retrieve in our middleware. If users want a catch all middleware I think each library would need to implement this. For example echo would need a middleware that uses the result of `c.Path()` as the route tag. 

We could provide a utility function that adds a passed route tag as an attribute and changes the span name as discussed previously.

Reply:
> We could provide a utility function that adds a passed route tag as an attribute and changes the span name as discussed previously.

I think this may be more flexible and powerful. Writing a little middleware that would use this function should not be a big deal.

Reply:
Added this utility function to otelhttp and route tagger middleware in 4 of the http router modules. Added a question about where to put the examples.

Reply:
Can you update the examples? The existing example structure is good enough. We can refine it later once we have a consensus on the implementation part.